bench/tiered/push_pull.c

```diff
@@ -311,16 +319,22 @@ run_test(const char *home, uint32_t num_records, bool flush, int counter)
  *     Populate the table.
  */
 static void
-populate(WT_SESSION *session, uint32_t num_records)
+populate(WT_SESSION *session, uint32_t num_records, uint32_t counter)
 {
     WT_CURSOR *cursor;
-    uint64_t i, str_len;
+    uint32_t w, z;
+    uint64_t v, i, str_len;
+
+    /* Seed the random number generator */
+    v = (uint32_t)getpid() + num_records + (2 * counter);
+    w = (uint32_t)(2 * flush) + num_records;
+    z = (2 * counter) + num_records;
 
-    srand((uint32_t)getpid() + num_records);
+    __wt_random_init_custom_seed(&rnd, v, w, z);
 
     str_len = sizeof(data_str) / sizeof(data_str[0]);
     for (i = 0; i < str_len - 1; i++)
-        data_str[i] = 'a' + (uint32_t)rand() % 26;
+        data_str[i] = 'a' + (uint32_t)__wt_random(&rnd) % 150;
```

Comment:
I'm still a worried/paranoid about compression here - this looks about ninety-some percent randomness, we could get more like 99.5% via:
```
data_str[i] = '\x01' + (uint8_t)__wt_random(&rnd) % 255;
```
Or close to 100% by using `value_format=u` and a totally random set of bytes 0x0 - 0x255.

The more important random issue is that we should be getting a completely different random string for every push of data.

The concern is not just theoretical.  Compressors are clever and getting cleverer - and if they "notice" that there's a range of values that is unused, they may take advantage of it.  For this 200 byte sequence, not using the lowest ascii values, allows for some (5-10%) compression.  Also if they notice that some 200 byte sequence was being used over and over, they'd make an "abbreviation" for it in the compression stream (it could be 2 bytes).  If a compressor gets even 5-10% more compaction than a competitor, they may become more favored (and by noticing the 200 byte repeated sequence, it might be more like 90+%).  Even if today's compressor doesn't notice it, some years from now, when AWS adopts the new compressor, our performance numbers on this benchmark may unexpectedly go up.  However, in MDB if we are storing fully encrypted and/or compressed objects to the cloud (likely) then our data will be 100% random, and MDB would not get the benefit.